src/WebJobs.Extensions.DurableTask/EntityScheduler/MessageSorter.cs

```diff
@@ -48,8 +57,13 @@ public void LabelOutgoingMessage(RequestMessage message, string destination, Dat
 
             DateTime timestamp = now;
 
+            bool countingEntries = false;
+            int destinationCount = 0;
+
             if (this.SendHorizon + reorderWindow + MinIntervalBetweenCollections < now)
             {
+                countingEntries = true;
+
```

Comment:
I realize this is pre-existing code, so just asking this for my own understanding: can you please clarify what each of terms in the if-statement mean? As in - what is the `SendHorizon`, `reorderWindow`, etc. Just trying to reason about this check. Thanks!

Reply:
the message sorter is designed to reorder/deduplicate messages. As a first approximation, it works like this:
- on sending side: for each destination, timestamp of last message sent. When sending a message, it includes that timestamp in the "Predecessor" field.
- on receiving side: for each source, timestamp of last message received. If it receives a message specifying "Predecessor" that matches what was recorded as last message, receive it and update the recorded timestamp. If it is smaller discard it as duplicate. If it is larger, put it into a buffer for out-of-order message until it matches.

now, this algorithm is modified so it stops making any reordering or deduplication efforts outside the specified tolerance window (default is 30 minutes). The `SendHorizon` is that window on the sender side, and the `ReceiveHorizon` is that window on the receiver side.

Reply:
And in order to not spend too much time going through the data structures to clean out irrelevant entries, we use `MinIntervalBetweenCollections`

Reply:
I see. That definitely allows me to understand the code better but I think I'm still missing a piece of the puzzle. I'd love to sync with you over the details of this later. For now, the changes make sense to me.